### PR TITLE
wayshot 1.3.0 -> 1.3.1

### PR DESCRIPTION
**Pull request by id3v1669**
## Description of changes

https://github.com/waycrate/wayshot/releases/tag/1.3.1

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

## Notes

Release version of package has faulty tests with `cargo test` but runs as intended, so `doCheck = false;` was added.

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/misc/wayshot/default.nix
Comment (CyberShadow) [inline pkgs/tools/misc/wayshot/default.nix:18]: Is there an upstream bug report?

Comment (id3v1669) [reply pkgs/tools/misc/wayshot/default.nix:18]: Did not find any report in source related to this issue. 
In developers makefile they do not run `cargo test`, just `cargo build`. So ignoring checks should not be an issue as they passed it to thier release version.

Comment (CyberShadow) [reply pkgs/tools/misc/wayshot/default.nix:18]: The upstream developers may appreciate a bug report about this, then.

It would also give us something to link to, so that we know what to do about disabling tests in the future. (E.g. if upstream fixes it, we can remove it; if upstream decides they don't care about tests, we will now know that.)

Comment (id3v1669) [reply pkgs/tools/misc/wayshot/default.nix:18]: Opened [issue](https://github.com/waycrate/wayshot/issues/113). What about pull request to nixpkgs? Should just wait for upstream developers response for the issue or do something else?

Comment (CyberShadow) [reply pkgs/tools/misc/wayshot/default.nix:18]: > What about pull request to nixpkgs? Should just wait for the upstream developers' response to the issue or do something else?

No need to wait, just add a link to the issue in a comment, that explains why tests are disabled.

**Comment by id3v1669** [comment]
Tests are disabled for now due to fail of `cargo test`, will be fixed in further updates.
Issue: [#113](https://github.com/waycrate/wayshot/issues/113)

File: pkgs/tools/misc/wayshot/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/misc/wayshot/default.nix:25]:
```suggestion
    maintainers = with maintainers; [ dit7ya id3v1669 ];
```

**Comment by SuperSandro2000** [comment]
We can fetchpatch the fix

**Comment by id3v1669** [comment]
> We can fetchpatch the fix

If you mean fetchpatch from commit. That won't be an option as commit has more fixes than version 1.3.1 needs.
Local patch can be created, but is it necessary just for one version?

**Comment by teto** [comment]
Result of `nixpkgs-review pr 302017` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>wayshot</li>
  </ul>
</details>

**Comment by id3v1669** [comment]
Result of `nixpkgs-review pr 302017` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wayshot</li>
  </ul>
</details>

**Comment by id3v1669** [comment]
> Result of `nixpkgs-review pr 302017` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 1 package failed to build:
> 
>     * wayshot
Just ran `nixpkgs-review pr --post-result 302017`, no issues.
Any ideas why that might happen?

**Comment by teto** [comment]
@ofborg build wayshot

**Comment by id3v1669** [comment]
> @ofborg build wayshot

Looks like now it shows different cargoHash from what was before. Please try again, I've changed to the one from ofborg logs.

**Comment by id3v1669** [comment]
@ofborg build wayshot

**Review by Ma27 — DISMISSED** [review]
Package looks good functionality-wise.

Please squash your commits together into the following two chunks:
* `maintainers: add id3v1669`
* `wayshot: 1.3.0 -> 1.3.1`

Also, you can pass arguments to `cargo test` via `checkFlags = [ ... ];`. Maybe that can be used to skip only the tests that are actually broken?

**Comment by id3v1669** [comment]
> Package looks good functionality-wise.
> 
> Please squash your commits together into the following two chunks:
> 
>     * `maintainers: add id3v1669`
> 
>     * `wayshot: 1.3.0 -> 1.3.1`
> 
> 
> Also, you can pass arguments to `cargo test` via `checkFlags = [ ... ];`. Maybe that can be used to skip only the tests that are actually broken?

Done. 
Regarding testing, I would agree with you if developers actually used some of those tests, but right now, they are irrelevant for this release.
Check please [this message](https://github.com/waycrate/wayshot/pull/114#issuecomment-2041169730), pr for integration testing is still WIP.

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**
